tests/integration/api/test_api_base.py

```diff
@@ -164,3 +164,24 @@ def test_hard_delete_record(base_app, db, create_record, create_pidstore):
     assert record is None
     assert record_pids is None
     assert record_identifier is None
+
+
+def test_redirect_records(base_app, db, create_record):
```

Comment:
I'd add also a test if ``redirect`` fails the object doesn't change (unhappy)

Reply:
So only thing this could fail is that we pass records without any pids. So do you think this `redirect` should also handle it.